src/app/api/collections/quality_matrix.py

```diff
@@ -43,6 +44,15 @@ class QualityMatrix(BaseModel):
     rows: list[QualityMatrixRow]
 
 
+def quality_matrix(collection: Tree, mode: QualityMatrixMode) -> QualityMatrix:
+    if mode == "replication-source":
```

Comment:
The usage of Literal forces you to use similar strings here, i.e., at multiple points. Please use an Enum, to DRY. I see no specific reason to not use an enum here.

Reply:
See comment below.